lib/puppet/provider/package/dnf.rb

```diff
@@ -30,7 +30,8 @@
 
   defaultfor :operatingsystem => :fedora
   notdefaultfor :operatingsystem => :fedora, :operatingsystemmajrelease => (19..21).to_a
-  defaultfor :osfamily => :redhat, :operatingsystemmajrelease => ["8"]
+  defaultfor :osfamily => :redhat
+  notdefaultfor :osfamily => :redhat, :operatingsystemmajrelease => ["6","7"]
```

Comment:
We still support [redhat 5](http://yum.puppet.com/puppet6/puppet6-release-el-5.noarch.rpm), so we should add it here.